graphql/resolvers.js

```
@@ -115,6 +116,27 @@ const resolvers = {
             return newBeacon;
         },
 
+        createGroup: async (_, { group }, { user }) => {
+            console.log(group);
+            for (let i = 0; i < 2; i++) {
+                try {
+                    const groupDoc = new Group({
+                        leader: user.id,
+                        shortcode: nanoid(),
+                        ...group,
+                    });
+                    const newGroup = await groupDoc.save().then(g => g.populate("leader"));
+                    user.groups.push(newGroup.id);
+                    await user.save();
+                    console.log(newGroup);
+                    return newGroup;
+                } catch (e) {
+                    console.log(e);
```

you probably want to use console.error, and you only want to silently catch it in case it is the duplicate error, otherwise we should rethrow the error. Here is an example on how you can do that, https://github.com/VandyHacks/witness/blob/main/pages/api/preadd.ts#L38 (I had to do something similar recently)

---

Done! Thanks for how to do that part!